File: Mars rober/rober.cpp
#include <vector>
#include <math.h>
#include <iostream>
#include <string>
#include <iomanip>
#include <sstream>


#define PI 3.1415926535


using namespace std;

int main (int argc, char *argv[])
{
    string input = "9.53 10 -1.00 1.00 -2.00 2.00 3.00 -3.00 4.00 4.00 5.00 -5.00 6.00 6.00 7.00 7.00 -8.00 8.00 9.00 9.00 10.00 -10.00";
    stringstream s_input (input);
    string value;
    float base_Angle=0;
    float x_Total=0, y_Total=0;

    getline (s_input, value, ' ');
    float wheelBase=stof(value);

    getline (s_input, value, ' ');
    int num_Movements=stoi(value);

    for (int n=0;n<num_Movements;n++)
    {
        getline (s_input, value, ' ');
        float distance=stof(value);
        
        getline (s_input, value, ' ');
        float steeringAngle=stof(value);

        

        float turnRadius;
        if (steeringAngle>0)
            turnRadius=wheelBase/sin((steeringAngle*PI)/180);
        else if (steeringAngle<0)
        {
            turnRadius=wheelBase/sin((-steeringAngle*PI)/180);
        }
        else
        {
            //cout<<std::fixed<<std::setprecision(2)<<"0.00 "<<distance<<' '<<0.00;
            x_Total+=distance*sin(base_Angle);
            y_Total+=distance*cos(base_Angle);
            continue;
        }
        
            

        float radRideAngle =distance/turnRadius;
        float degRideAngle = (180*(radRideAngle))/PI;
        if(degRideAngle>360)
        {
            int vueltas=degRideAngle/360;
            degRideAngle=degRideAngle-360*vueltas;
        }
        if (degRideAngle<0)
        {
            degRideAngle=360+degRideAngle;
        }
        if (steeringAngle<0)
        {
            degRideAngle=360.0-degRideAngle;
        }
        float x,y;
        if(steeringAngle>0)
        {
            x=turnRadius-(turnRadius*cos(radRideAngle));
            y=turnRadius*sin(radRideAngle);
        }
        else
        {
            x=(turnRadius*cos(radRideAngle))-turnRadius;
            y=turnRadius*sin(radRideAngle);
        }

        float x_New=cos(base_Angle)*x+sin(base_Angle)*y;
        float y_New=-sin(base_Angle)*x+cos(base_Angle)*y;
        x_Total+=x_New;
        y_Total+=y_New;
        base_Angle+=(degRideAngle*PI)/180;
    }
    float final_Angle=(base_Angle*180)/PI;
    if ((int)final_Angle>=360)
    {
        int vueltas=final_Angle/360;
        final_Angle-=360*vueltas;
    }
    if(final_Angle<0)
        final_Angle=360-final_Angle;
    
    cout<<std::fixed<<std::setprecision(2)<<x_Total<<' '<<y_Total<<' '<<final_Angle;
}

File: rober ptyhon/rober.py
import requests
import math
def mover (x, y ,angle,distance, steeringAngle) :
    if steeringAngle>0 :
        turnRadius=wheelBase/math.sin((steeringAngle*PI)/180)
    elif steeringAngle <0 :
        turnRadius=wheelBase/math.sin((-steeringAngle*PI)/180)
    else :
        x+=distance*math.sin(angle)
        y+=distance*math.cos(angle)
        return x, y , angle
    radRideAngle =distance/turnRadius
    degRideAngle = (180*(radRideAngle))/PI
    if degRideAngle>360 :
        vueltas=int (degRideAngle/360)
        degRideAngle=degRideAngle-360*vueltas
    if degRideAngle<0 :
        degRideAngle=360+degRideAngle
    if steeringAngle<0 :
        degRideAngle=360.0-degRideAngle
    if steeringAngle>0 :
        x=turnRadius-(turnRadius*math.cos(radRideAngle))
        y=turnRadius*math.sin(radRideAngle)
    else :
        x=(turnRadius*math.cos(radRideAngle))-turnRadius
        y=turnRadius*math.sin(radRideAngle)
    x_New=math.cos(angle)*x+math.sin(angle)*y
    y_New=-math.sin(angle)*x+math.cos(angle)*y
    x+=x_New
    y+=y_New
    angle+=(degRideAngle*PI)/180
    if angle>=360 :
        vueltas=angle/360
        angle-=360*vueltas
    if(angle<0) :
        angle=360-angle
    return x,y,angle
PI=3.1415926535
url_core='https://rover.codingcontest.org'
s_input='L5_MAF3401R'
username='xivitazo'
resp=requests.get(url_core+'/rover/create?map='+s_input+'&username='+username+'&contestId=practice')
UUID=resp.text
#print (UUID)
spetifications=requests.get(url_core+'/rover/'+UUID)
print (spetifications.text)
splited=spetifications.text.split()
wheelBase = float(splited[0])
maxSteeringAngle = float(splited[1])
targetX = float(splited[2])
targetY = float(splited[3])
targetRadius = float(splited[4])

turn_radius=math.sqrt(math.pow(targetX,2)+math.pow(targetY,2))

steeringAngle = ((math.asin(wheelBase/turn_radius))*180)/PI
#minTurnRadius=wheelBase/math.sin((maxSteeringAngle*PI)/180)

#print (str(turn_radius )+ ' ' + str(minTurnRadius))



toMoveDistance=math.atan(targetY/targetX)*turn_radius



status=requests.get(url_core+'/rover/move/'+UUID+'?distance='+str(toMoveDistance)+'&steeringAngle='+str(steeringAngle))
print(status.text)
splited=status.text.split()
if splited[0]== 'OK' :
    movedDistance=float(splited[1])
elif splited [0] == 'ERROR':
    print(status.text)
    exit()
elif splited[0] == 'PASS':
    passKey = splited[1]
    totalDistance = splited [2]
    endPass=requests.get(url_core+'/rover/move/'+UUID+'?distance='+str(toMoveDistance)+'&steeringAngle='+str(steeringAngle))
    splited=endPass.text.split()
    passKey = splited[1]
    totalDistance = splited [2]
    print (passKey)
    exit ()

print (str(mover(0,0,0,toMoveDistance,steeringAngle)) + ' '+ str(toMoveDistance) + ' '+ str(steeringAngle))